src/main/play-doc/developer/SignalingApplicationState.md

```diff
@@ -39,3 +39,9 @@ StatusService.signalStartedOrExit()
 ```
 
 After reading the following sections, you may also wish to refer to [the reference documentation for conductr-bundle-lib](https://github.com/typesafehub/conductr-bundle-lib#typesafe-conductr-bundle-library).
+
+## Application Exit Codes
+
+ConductR uses your application's exit code to determine whether it should be restarted in order to reform the cluster. Following POSIX conventions, applications with non-zero exit codes are considered failed and will be restarted, while applications exiting with exit code 0 will be viewed as having exited normally, and thus ConductR will take no action.
+
+This is particularly important for Akka applications using [Split Brain Resolver](http://doc.akka.io/docs/akka/rp-15v09p01/scala/split-brain-resolver.html) with the [registerOnMemberRemoved](http://doc.akka.io/docs/akka/rp-15v09p01/scala/split-brain-resolver.html#Strategies) callback to shut down the ActorSystem after a split brain downing decision has been made. A non-zero return code must be issued to ensure that ConductR is instructed to recover the cluster to its previous state.
```

Comment:
Afaik Akka SBR provides no signal other than removing the current member. Therefore if the current member is downed, and your process provides no explicit means for it to be downed, then you should return a non zero exit code. 

Perhaps a new member state indicating an auto removal would be good for the future. Wdyt @patriknw?

> On 25 Aug 2016, at 07:23, Jamie Longmuir <notifications@github.com> wrote:
> 
> In src/main/play-doc/developer/SignalingApplicationState.md:
> 
> > @@ -39,3 +39,9 @@ StatusService.signalStartedOrExit()
> >  ```
> >  
> >  After reading the following sections, you may also wish to refer to [the reference documentation for conductr-bundle-lib](https://github.com/typesafehub/conductr-bundle-lib#typesafe-conductr-bundle-library).
> > +
> > +## Application Exit Codes
> > +
> > +ConductR uses your application's exit code to determine whether it should be restarted in order to reform the cluster. Following POSIX conventions, applications with non-zero exit codes are considered failed and will be restarted, while applications exiting with exit code 0 will be viewed as having exited normally, and thus ConductR will take no action.
> > +
> > +This is particularly important for Akka applications using [Split Brain Resolver](http://doc.akka.io/docs/akka/rp-15v09p01/scala/split-brain-resolver.html) with the [registerOnMemberRemoved](http://doc.akka.io/docs/akka/rp-15v09p01/scala/split-brain-resolver.html#Strategies) callback to shut down the ActorSystem after a split brain downing decision has been made. A non-zero return code must be issued to ensure that ConductR is instructed to recover the cluster to its previous state.
> Thanks - it's not clear to me how (or if it's possible) to distinguish between a graceful exit (ie - for a planned upgrade, scaling down, etc) where a return code of 0 makes sense with ConductR not restarting the service, versus an SBR induced downing where we would expect ConductR to restart the service.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>